### DISCUSSION TITLE
Time independent unknown parameters in the convection-diffusion equation

### DISCUSSION THREAD

**Discussion by j6k5tki** [lululxvi/deepxde#1115] [Q&A] [answered]
Hello, Dr. Lu,

I have been working on obtaining time-independent parameters (`D(r), V(r)`) in an inverse problem for the 1D diffusion equation. However, the unknown parameters in my case are **space-dependent** and **time-independent**, which differs from the example codes. I am trying to set constraints to achieve these parameters that satisfy the time-independent conditions. (For the time dependent case, I refer to #761 to implement unknown parameters that vary temporally and spatially.)
I have implemented a code to obtain these parameters, but I only achieve the temporally varying parameters despite adjusting the weights of the residuals. Are there any other methods or techniques I could use to achieve this goal?

Thank you for your time.

#
`def pde(x, y, Q):`

      r = x[:,0:1] # space
      t = x[:,1:2] # time
  
      n = y[:,0:1] # particle density
      D = y[:,1:2] # diffusion coefficient
      V = y[:,2:3] # convection velocity
  
      TAU  = 100/(1+tf.exp(700*(r - a))) + tau
  
      dn_t  = dde.grad.jacobian(y, x, i=0, j=1)
      dn_r  = dde.grad.jacobian(y, x, i=0, j=0)
      dn_rr = dde.grad.hessian(y, x, component=0, i=0, j=0)
      dD_r  = dde.grad.jacobian(y, x, i=1, j=0)
      dV_r  = dde.grad.jacobian(y, x, i=2, j=0)
  
      res = r*dn_t + r*n/TAU - r*Q - ((D*dn_r - V*n) + r*(dD_r*dn_r +D*dn_rr - dV_r*n - V*dn_r))

      # time_dep == False (for the time indep. sol.)
      if time_dep:
          return res
      else: 
          dD_t = dde.grad.jacobian(y, x, i=1, j=1)
          dV_t = dde.grad.jacobian(y, x, i=2, j=1)
          return [res, dD_t, dV_t]

**Comment by JaehongPark-Plasma** [comment]
I also want to get time-independent parameters in the inverse problem!!! :)

**Comment by lululxvi** [comment, marked as answer]
See FAQ "Q: Solve inverse problems with unknown parameters/fields in the PDEs or initial/boundary conditions."

**Reply by j6k5tki** [reply]
Thank you @lululxvi . I solve the problem.
